### PR TITLE
Removed shell scripts, changed update functionality

### LINKED ISSUE #148: []version is bugged
FIX IT CODE

### PR DESCRIPTION
**Pull request by MetalBreaker**
Works on every platform Git and .NET Core do. Closes #148

### CODE REVIEW

**Comment by MetalBreaker** [comment]
Whoops. Forgive me for those meaningless commits. I thought I was committing them to my "testupdate" branch.

**Comment by MetalBreaker** [comment]
Not quite cooked yet.

**Comment by YilianSource** [comment]
Why did you close this? Does it not work?

**Comment by MetalBreaker** [comment]
As I said, it isn't quite cooked enough yet. Some things should be done better. Give me a few days.

**Comment by MetalBreaker** [comment]
Should be working now!

**Comment by MetalBreaker** [comment]
Should be fully working and reliable now.

File: Helpers/ProcessHelper.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        }
+
+        public static void KillAndDispose(this Process process)
+        {
+            process.Kill();
+            process.Dispose();
+        }
+        public static void RedirectToConsoleOutput(this Process process)
+        {
+            _ = process.StandardOutput.BaseStream.CopyToAsync(Console.OpenStandardOutput());
```
Comment (CodeMyst) [inline Helpers/ProcessHelper.cs]: `Stream` has a non async [`CopyTo`](https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.copyto?view=netstandard-2.0#System_IO_Stream_CopyTo_System_IO_Stream_)

File: BrackeysBot.cs [hunk 2]
```diff
@@ -21,14 +22,17 @@ public sealed class BrackeysBot
...
         public const int Version = 3;
 
+        public bool OriginalProcess = true;
+
         public DataModule Data { get; set; }
         public CommandHandler Commands { get; set; }
 
         private IServiceProvider _services;
-        private DiscordSocketClient _client;
+        public DiscordSocketClient _client;
```
Comment (CodeMyst) [inline BrackeysBot.cs]: No need to make this public. Also you made it public but now has the wrong naming style.

File: BrackeysBot.cs [hunk 3]
```diff
@@ -113,25 +117,38 @@ public async Task Start()
...
-        public async Task ShutdownAsync(bool terminate)
+        public Task Shutdown(bool terminate)
         {
-            await _client.LogoutAsync();
             if (terminate)
-            {
-                Environment.Exit(0);
-            }
+                _client.Disconnected += Terminate;
+            _ = _client.StopAsync();
```
Comment (CodeMyst) [inline BrackeysBot.cs]: Document why you need to use `_` to discard.

File: BrackeysBot.cs [hunk 4]
```diff
@@ -1,3 +1,4 @@
+#pragma warning disable CS1998
```
Comment (CodeMyst) [inline BrackeysBot.cs]: Don't do this.

File: Commands/VersioningCommand.cs [hunk 5]
```diff
@@ -3,6 +3,7 @@
 using System.Diagnostics;
 using System.Threading.Tasks;
 using System.Runtime.InteropServices;
+using static BrackeysBot.ProcessHelper;
```
Comment (YilianSource) [inline Commands/VersioningCommand.cs]: I would prefer if you'd not use this, just makes the file less readable.

File: BrackeysBot.cs [hunk 6]
```diff
@@ -113,25 +117,38 @@ public async Task Start()
         /// <summary>
         /// Asynchronously logs out the bot. Also terminates the application, if specified.
         /// </summary>
-        public async Task ShutdownAsync(bool terminate)
+        public Task Shutdown(bool terminate)
```
Comment (CodeMyst) [inline BrackeysBot.cs]: Can just make it `void`

File: BrackeysBot.cs [hunk 7]
```diff
@@ -21,14 +22,17 @@ public sealed class BrackeysBot
         public static IConfiguration Configuration { get; set; }
         public const int Version = 3;
 
+        public bool OriginalProcess = true;
```
Comment (CodeMyst) [inline BrackeysBot.cs]: Should be a property.

**Comment by CodeMyst** [comment]
It also has a lot of inconsistent code style. Follow the project's code style and naming style.

File: BrackeysBot.cs [hunk 8]
```diff
@@ -113,25 +117,38 @@ public async Task Start()
         /// <summary>
         /// Asynchronously logs out the bot. Also terminates the application, if specified.
```
[the remark this answers is not in the document]
Comment (MetalBreaker) [reply BrackeysBot.cs]: It is async.

[the remark this answers is not in the document]
Comment (CodeMyst) [reply BrackeysBot.cs]: It's not marked as async.

**Comment by MetalBreaker** [comment]
Alright. I'll be rewriting the whole thing, looks like. Not enough time on my hands right now.
Closing.

**Pull request closed without merging**